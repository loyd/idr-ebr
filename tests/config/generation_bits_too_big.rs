use idr_ebr::{Config, Idr};

struct InvalidConfig;
impl Config for InvalidConfig {
    const MAX_PAGES: u32 = 26;
}

fn main() {
    let _ = Idr::<u64, InvalidConfig>::new();
}
