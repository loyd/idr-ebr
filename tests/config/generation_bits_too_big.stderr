error[E0080]: evaluation of `<InvalidConfig as idr_ebr::config::ConfigPrivate>::ENSURE_VALID` failed
 --> src/config.rs
  |
  |         assert!(Self::GENERATION_BITS <= 32);
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'assertion failed: Self::GENERATION_BITS <= 32', $DIR/src/config.rs:81:9
  |
  = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

note: the above error was encountered while instantiating `fn idr_ebr::Idr::<u64, InvalidConfig>::new`
 --> tests/config/generation_bits_too_big.rs:9:13
  |
9 |     let _ = Idr::<u64, InvalidConfig>::new();
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
